linode/manager
# upcoming: [DI-20595] - ACLP Supported regions per service type

## Description 📝

ACLP supports a limited set of regions per service type and capability based on LaunchDarkly config. Regions are now correctly filtered according to the capability they belong to and the supported regions fetched from launchDarkly.

## Changes  🔄

1. Added new key for supported regions in aclpResourceType Flag.
2. Validated the region ids from flag with all the regions fetched from api.
3. Filtered the service type specific regions using corresponding capabilities.
4. Added relevant unit test to verify.

## Target release date 🗓️

12 December 2024

## Preview 📷

<img src="https://github.com/user-attachments/assets/bdc5bccf-47c8-4cb8-8d9b-e02b328b4747" width = "300" height = "200">


## How to test 🧪

1. Navigate to monitor tab.
2. Select a dashboard.
3. Open networks tab, find the launchDarkly config, see supported region ids in aclpResourceTypeMap, verify that these are same as the region ids listed in 'regions' autocomplete popper list. In case supportedRegionIds is null/undefined, all regions per capability will be shown.
Note: ids are listed at the end of region option label in autocomplete list.


<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅


</details>

---
